src/main/java/entity/BeginAndDueDates.java

```diff
@@ -7,8 +7,8 @@
  * This class is immutable to ensure data integrity.
  */
 public class BeginAndDueDates {
-    private final LocalDate beginDate;
-    private final LocalDate dueDate;
+    private LocalDate beginDate;
+    private LocalDate dueDate;
```

Comment:
```suggestion
    private final LocalDate beginDate;
    private final LocalDate dueDate;
```
Entities must be immutable. Removing ```final``` keywords violates this fundamental Clean Architecture principle and compromises data integrity.